indico/htdocs/js/indico/jquery/itempicker.js

```diff
@@ -0,0 +1,125 @@
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function($) {
+    $.widget('indico.itempicker', {
+
+        options: {
+            onSelect: null, // function called every time an item gets selected from the list
+            onActionButtonClicked: null,
+            createItem: null, // function returning a jQuery object representing each item in the dropdown
+            items: [], // all the options that should be included in the dropdown
+            containerClasses: '' // extra classes to be applied to the container
+        },
+
+        _create: function() {
+            var self = this;
+            var element = self.element;
+            var items = self.options.items.length ? self.options.items : (element.data('items') || []);
+            var dropdownContainer = $('<div>', {'class': 'dropdown-container'});
+            var filterWrapper = $('<div>', {'class': 'dropdown-filter-wrapper'});
+            var filterInput = $('<input>', {'type': 'text', 'class': 'filter-input'});
+            var itemsContainer = $('<div>', {'class': 'dropdown-items-container'});
+            var handleInput = function() {
+                var textTyped = filterInput.val().trim().toLowerCase();
+
+                dropdownContainer.find('.dropdown-item').each(function() {
+                    var $item = $(this);
+                    $item.toggle($item.data('filter').toLowerCase().indexOf(textTyped) !== -1);
+                });
+            };
+
+            filterWrapper.append(filterInput);
+            filterInput.clearableinput({onInput: handleInput, onClear: handleInput})
+                       .clearableinput('initSize', '1rem', '1.8em');
+
+            dropdownContainer.append(filterWrapper).append($('<div>', {'class': 'item-picker-divider'}));
+
+            $.each(items, function(index, value) {
+                var item;
+                if ($.isFunction(self.options.createItem)) {
+                    item = self.options.createItem.call(element, value);
+                } else {
+                    item = $('<div>', {'class': 'dropdown-item', 'data': {'filter': value.title},
+                                       'text': value.title});
+                }
+
+                item.on('click', function() {
+                    var $this = $(this);
+                    if ($.isFunction(self.options.onSelect)) {
+                        self.options.onSelect.call(element, value);
+                        element.qtip('hide');
+                    }
+                });
+
+                itemsContainer.append(item);
+            });
+
+            dropdownContainer.append(itemsContainer);
+
+            if (element.data('with-action-button')) {
```

Comment:
why use a data attribute for this? right now you *have to* mix data attribute and arguments passed to the widget constructor